src/transformers/generation/utils.py

```diff
@@ -689,12 +689,14 @@ def _update_model_kwargs_for_generation(
                     dim=-1,
                 )
 
-        if (
-            model_kwargs.get("use_cache", True)
-            and "cache_position" in model_kwargs
-            and model_kwargs["cache_position"] is not None
-        ):
+        if model_kwargs.get("use_cache", True):
```

Comment:
Changes in this file copied from the [PR](https://github.com/huggingface/transformers/pull/31898) that updates `prepare_inputs_for_generation` (they are also needed for this PR)